Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -266,7 +266,7 @@ bool parse_it(Iterator &first, Iterator last, RDKit::RWMol &mol) {
       } else {
         if (!parse_atom_labels(first, last, mol)) return false;
       }
-    } else if ((first + 9) < last &&
+    } else if (std::distance(first, last) > 9 &&
```

Comment:
It might make sense to add

```
while (...)
++first;
Iterator::difference_type length = std::distance(first, last);

if (*first == '(') {
      if (!parse_coords(first, last, mol)) return false;
    } else if (*first == '$') {
       if ( (length > 4) ... 
```

 in the while loop and use length instead of the calculations.  This bug may affect the other distance calculations.